kernel/nativevalues.ml

```diff
@@ -120,15 +120,15 @@ let accumulate_tag = 0
 (** Unique pointer used to drive the accumulator function *)
 let ret_accu = Obj.repr (ref ())
 
-type accu_val = { mutable acc_atm : atom; acc_arg : Obj.t list }
+type accu_val = { mutable acc_atm : atom; acc_arg : t list }
 
 external set_tag : Obj.t -> int -> unit = "coq_obj_set_tag"
 
 let mk_accu (a : atom) : t =
   let rec accumulate data x =
     if x == ret_accu then Obj.repr data
     else
-      let data = { data with acc_arg = x :: data.acc_arg } in
+      let data = { data with acc_arg = (Obj.magic x : t) :: data.acc_arg } in
```

Comment:
I don't think you need this `Obj.magic`. Just change the test above into `Obj.repr x == ret_accu` to make the function well-typed.